(baby-project-deployment) #3 Creates a simple deployment action (#5)
TurkuForge/baby-project-deployment: README.md

# Baby Project Deployment
We use a micro repo structure for a few reasons, but the main one is that we can make use of

> GitHub pages for each repo separately. That way we can deploy documents/our documentation, test coverage or even a full webapp without extra fees. 

Reviewer: ```suggestion
GitHub pages for each repo separately. That way we can deploy documents, test coverage or even a full webapp without extra fees. 
```
or
```suggestion
GitHub pages for each repo separately. That way we can deploy our documentation, test coverage or even a full webapp without extra fees. 
```

# Baby Project Deployment
We use a micro repo structure for a few reasons, but the main one is that we can make use of

> GitHub pages for each repo separately. That way we can deploy docs, test coverage or even a full webapp without extra

Reviewer 2: ```suggestion
GitHub pages for each repo separately. That way we can deploy docs, test coverage or even a full webapp without
```
I belive you have an extra _extra_ in this sentence :D

> GitHub pages for each repo separately. That way we can deploy docs, test coverage or even a full webapp without extra

Reviewer: Same opinion as Otto, I think it's better to avoid abbreviations when not needed. 😄

```suggestion
GitHub pages for each repo separately. That way we can deploy documents/our documentation, test coverage or even a full webapp without extra
```

> GitHub pages for each repo separately. That way we can deploy our documentation, test coverage or even a full webapp without extra fees. 

Reviewer: ```suggestion
GitHub pages for each repo separately. That way we can deploy our documentation, test coverage, or even a full web app without extra fees. 
```

dealing with extra fees. That comes with some drawbacks, but some benefits as well. We publish individual releases from
each repo and use this repo to deploy them.

> We use [Dokku](https://dokku.com/) for deployment. The main reason is that it's easy to set up and quite easy to scale and 

Reviewer: ```suggestion
We use [Dokku](https://dokku.com/) for deployment. The main reason is that it's quite easy to set up, scale and 
```

> We use [Dokku](https://dokku.com/) for deployment. The main reason is that it's quite easy to set up, scale and 
> maintain. You can however deploy it locally as well using Docker.

Reviewer: ```suggestion
We use [Dokku](https://dokku.com/) for deployment. The main reason is that it's quite easy to set up, scale, and maintain. You can however deploy it locally as well using Docker.
```

> maintain. You can however deploy it locally as well using docker.

Reviewer: ```suggestion
maintain. You can however deploy it locally as well using Docker.
```

First build it 🛠
```shell

> docker build -t baby-project .

Reviewer: I think it's always better to write the long form of the flags in scripts since it's easier for future readers to grasp what the commands do. E.g. if I don't know/remember what `-t` does, it reads more clearly if it says:
```suggestion
docker build --tag list baby-project .
```

Author: What is the `list` there for?

Reviewer: I thought it was the long-form of `-t`. Is it only an argument or something? Not sure actually. 🤔
![image](https://user-images.githubusercontent.com/17124245/118764664-8ca7d280-b882-11eb-9b3c-1ae5948f7e69.png)

Reviewer: Aaah, it's probably the type of the argument, `string`, `list`, `bytes`, etc. 😄
So should not be included in the long-form version. 👍

Author: Yea I removed it in the readme because I saw it worked without it

```
Then run it 🏃‍
```shell

> docker run -dp 8080:80 baby-project

Reviewer: Interesting! Didn't know there can be spaces in a long-form version of a flag, at least with `docker`.
```suggestion
docker run --detach --publish list 8080:80 baby-project
```
![image](https://user-images.githubusercontent.com/17124245/118599853-d4fabe00-b7b8-11eb-9e53-c260144399da.png)

Author: Again what is the `list` for?

Reviewer: https://github.com/TurkuForge/baby-project-deployment/pull/5#discussion_r634940958

```
Then go to 
`localhost:8080`

> and then you have the production env running locally.

Reviewer 2: ```suggestion
and then you have the production environment running locally.
```
Might be good to spell out `environment`

> The official deployment to the server is done through GitHub actions. The prosess is basicly just bumping the version in

Reviewer: > The prosess is basicly just bumping the version in the `.env` file

Where does it do this bumping in the code?

Author: Well, you as the person updating will be bumping the version.

> The official deployment to the server is done through GitHub actions. The prosess is basicly just bumping the version in

Reviewer: ```suggestion
The official deployment to the server is done through GitHub actions. The process is basically just bumping the version in
```

> the `.env` file that we keep in the version control and create a PR. Only orginization mebers can create PRs for this repo.

Reviewer: ```suggestion
the `.env` file that we keep in the version control and create a PR. Only organization members can create PRs for this repo.
```
